price-estimator/src/filter.rs

```diff
@@ -24,40 +20,32 @@ pub fn all(
 ) -> impl Filter<Extract = impl Reply, Error = Infallible> + Clone {
     warp::path!("api" / "v1" / ..)
         .and(
-            markets(orderbook.clone())
+            markets(orderbook.clone(), token_info.clone())
                 .or(estimated_buy_amount(orderbook.clone(), token_info.clone()))
-                .or(estimated_amounts_at_price(orderbook.clone(), token_info))
-                .or(estimated_best_ask_price(orderbook)),
+                .or(estimated_amounts_at_price(
+                    orderbook.clone(),
+                    token_info.clone(),
+                ))
+                .or(estimated_best_ask_price(orderbook, token_info)),
         )
         .recover(handle_rejection)
 }
 
-#[derive(Debug)]
-struct NoTokenInfo;
-impl Reject for NoTokenInfo {}
-
 async fn handle_rejection(err: Rejection) -> Result<impl Reply, Infallible> {
-    let code;
-    let message;
-
-    if err.is_not_found() {
-        code = StatusCode::NOT_FOUND;
-        message = "invalid url path";
-    } else if let Some(NoTokenInfo) = err.find() {
-        code = StatusCode::BAD_REQUEST;
-        message = "request with atoms=true for token we don't have erc20 info for";
-    } else if let Some(InternalError(err)) = err.find() {
-        log::warn!("internal server error: {:?}", err);
-        code = StatusCode::INTERNAL_SERVER_ERROR;
-        message = "internal server error";
+    let (code, message) = if let Some(reason) = err.find::<RejectionReason>() {
+        log::warn!("rejection reason: {:?}", reason);
+        reason.as_http_error()
+    } else if err.is_not_found() {
+        (StatusCode::NOT_FOUND, "invalid url path")
```

Comment:
Can we log the requests here and in the BAD_REQUEST case as well? I was seeing some 404 in the logs but wasn't able to find out the exact qeuery that did them.

Reply:
So I think the warp log filter should log all requests currently. Its probably a logging configuration issue in the deployment.

Reply:
Addressed by #1302 and #1303